packages/babel-core/src/transformation/file/index.js

```diff
@@ -410,8 +410,22 @@ export default class File extends Store {
   }
 
   parse(code: string) {
+    let opts = this.opts;
+    let parseCode = parse;
+
+    if (opts.parserOpts.recast) {
```

Comment:
Is `parserOpts` always defaulted to an empty array? If not, then `opts.parserOpts && opts.parserOpts.recast` is safer. Not sure if destructuring would throw if `parserOpts` is undefined `let {parserOpts: {recast}} = this.opts;`